File: Frontend/GestionBeneficiarioFrontend/src/utils/mappers/beneficiario.mapper.ts
import type { BeneficiarioResponseDtoApi } from "../../types/api/beneficiario.api";
import type { Beneficiario } from "../../types/ui/beneficiario.ui";

export const mapBeneficiarioApiToUi = (b: BeneficiarioResponseDtoApi | any): Beneficiario => ({
  id: b.Id ?? b.id,
  nombres: b.Nombres ?? b.nombres,
  apellidos: b.Apellidos ?? b.apellidos,
  documentoIdentidadId: b.DocumentoIdentidadId ?? b.documentoIdentidadId,
  numeroDocumento: b.NumeroDocumento ?? b.numeroDocumento,
  fechaNacimiento: b.FechaNacimiento ?? b.fechaNacimiento,
  sexo: ((b.Sexo ?? b.sexo) === "M" ? "M" : "F"),
  documentoNombre: b.DocumentoNombre ?? b.documentoNombre ?? null,
  documentoAbreviatura: b.DocumentoAbreviatura ?? b.documentoAbreviatura ?? null,
});

File: Backend/GestionBeneficiario_BackendAPI/GestionBeneficiario_BackendAPI/Data/Mapper/BeneficiarioMapper.cs
using GestionBeneficiario_BackendAPI.Models;
using Microsoft.Data.SqlClient;

namespace GestionBeneficiario_BackendAPI.Data.Mapper
{
    public class BeneficiarioMapper
    {
        public static Beneficiario Map(SqlDataReader reader)
        {
            return new Beneficiario
            {
                Id = reader.GetInt32(reader.GetOrdinal("Id")),
                Nombres = reader.GetString(reader.GetOrdinal("Nombres")),
                Apellidos = reader.GetString(reader.GetOrdinal("Apellidos")),
                DocumentoIdentidadId = reader.GetInt32(reader.GetOrdinal("DocumentoIdentidadId")),
                NumeroDocumento = reader.GetString(reader.GetOrdinal("NumeroDocumento")),
                FechaNacimiento = reader.GetDateTime(reader.GetOrdinal("FechaNacimiento")),
                Sexo = reader.GetString(reader.GetOrdinal("Sexo")) // char(1) funciona como string
            };
        }
    }
}
